test/e2e/util.go

```diff
@@ -780,3 +852,74 @@ func HighLatencyRequests(c *client.Client, threshold time.Duration) (int, error)
 
 	return len(badMetrics), nil
 }
+
+// Retrieve metrics information
+func getMetrics(c *client.Client) (string, error) {
+	body, err := c.Get().AbsPath("/metrics").DoRaw()
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+// Retrieve debug information
+func getDebugInfo(c *client.Client) (map[string]string, error) {
```

Comment:
A very simple example is we can see a large spike in the # of go-routines in cases where we expect it to hold steady.